shared/team-building/index.tsx

```diff
@@ -295,6 +296,10 @@ class TeamBuilding extends React.PureComponent<Props> {
     this.props.fetchUserRecs()
   }
 
+  componentWillUpdate = () => {
```

Comment:
Did you mean `componentDidUpdate`? `componentWillUpdate` is deprecated. Also shouldn't this only get called when some certain things change? Otherwise it seems you'd be relying on the reducer (to reuse the same reference) to prevent this from getting called indefinitely.

Reply:
Assuming that one understands what triggers a re-render here, useEffect might be useful

Reply:
I wanted to do `useEffect` but that only works with pure functional components.